### PR TITLE
Nim-1.2.2

### PR DESCRIPTION
**Pull request by stefantalpalaru**
This PR is being used to test the second Nim-1.2.2 release candidate in our CI.

### CODE REVIEW

File: config.nims [hunk 1]
```diff
@@ -58,3 +58,10 @@ if not defined(macosx):
 # `switch("warning[CaseTransition]", "off")` fails with "Error: invalid command line option: '--warning[CaseTransition]'"
 switch("warning", "CaseTransition:off")
 
+# The compiler doth protest too much, methinks, about all these cases where it can't
+# do its (N)RVO pass: https://github.com/nim-lang/RFCs/issues/230
+switch("warning", "ObservableStores:off")
```
Comment (arnetheduck) [inline config.nims:63]: we want to disable this switch locally, where it happens - it's quite a dangerous thing that severaly impacts how to reason about code

Comment (stefantalpalaru) [reply config.nims:63]: > we want to disable this switch locally, where it happens

Hundreds of times?

Comment (arnetheduck) [reply config.nims:63]: yes - the warning warns about unreliable behaviour in the face of exceptions, and we're using lots of those - rather go through the warnings and make a concious decision than risk hitting some weird condition at runtime in an exceptional case that we didn't test properly.

Comment (arnetheduck) [reply config.nims:63]: basically, the code where the warning is appearing should probably be ported to `newClone` - and we can add the warning disabling in there.

Comment (arnetheduck) [reply config.nims:63]: ping @zah

Comment (zah) [reply config.nims:63]: The warning seems buggy.

It reports a number of lines in nim-serialization that should be considered a classic case of C++-like safe NRVO:

```nim
var s = memoryOutput()
```

```nim
var writer = unpackArgs(init, [WriterType(Format), s, params])
```

Will ask Araq to look into it.

Comment (zah) [reply config.nims:63]: The following fix should improve things:
https://github.com/nim-lang/Nim/pull/14582

Comment (narimiran) [reply config.nims:63]: > The following fix should improve things: nim-lang/Nim#14582

This has now been backported.

Our nightlies builds will pick this up later today. If you want to test it ASAP, you can use this commit: https://github.com/nim-lang/Nim/tree/10fb986fb607b8b225cf86148479ccc20c79b9ac

Comment (narimiran) [reply config.nims:63]: A new nightly build is available at https://github.com/nim-lang/nightlies/releases/tag/2020-06-08-version-1-2-10fb986

Comment (stefantalpalaru) [reply config.nims:63]: Nim submodule bumped to the latest RC commit.

I commented out the warning block, locally, and got a list of remaining ObservableStores warnings when building the tools: https://gist.github.com/stefantalpalaru/f7c14f68672eb0f6211ffc84ab200cfb

Comment (zah) [reply config.nims:63]: I still see a lot of false-positives in there. Maybe there is not enough time to address this properly for 1.2.2 and we can indeed disable the warning for now.

**Pull request merged**